source/extensions/filters/network/common/redis/redis_command_stats.cc

```diff
@@ -0,0 +1,106 @@
+  if (enabled_) {
+    // Create StatName for each Redis command. Note that we don't include Auth or Ping.
+    for (const std::string& command :
+         Extensions::NetworkFilters::Common::Redis::SupportedCommands::simpleCommands()) {
+      createStats(command);
+    }
+    for (const std::string& command :
+         Extensions::NetworkFilters::Common::Redis::SupportedCommands::evalCommands()) {
+      createStats(command);
+    }
+    for (const std::string& command : Extensions::NetworkFilters::Common::Redis::SupportedCommands::
+             hashMultipleSumResultCommands()) {
+      createStats(command);
+    }
+    createStats(Extensions::NetworkFilters::Common::Redis::SupportedCommands::mget());
+    createStats(Extensions::NetworkFilters::Common::Redis::SupportedCommands::mset());
+  }
+}
+
+void RedisCommandStats::createStats(std::string name) {
+  stat_name_set_.add(name + ".total");
+  stat_name_set_.add(name + ".success");
+  stat_name_set_.add(name + ".error");
+  stat_name_set_.add(name + ".latency");
+}
+
+Stats::SymbolTable::StoragePtr RedisCommandStats::addPrefix(const Stats::StatName name) {
+  Stats::StatNameVec names_with_prefix;
+  names_with_prefix.reserve(2);
+  names_with_prefix.push_back(prefix_);
+  names_with_prefix.insert(names_with_prefix.end(), name);
+  return scope_.symbolTable().join(names_with_prefix);
+}
+
+Stats::Counter& RedisCommandStats::counter(std::string name) {
+  Stats::StatName stat_name = stat_name_set_.getStatName(name);
+  const Stats::SymbolTable::StoragePtr stat_name_storage = addPrefix(stat_name);
+  return scope_.counterFromStatName(Stats::StatName(stat_name_storage.get()));
+}
+
+Stats::Histogram& RedisCommandStats::histogram(std::string name) {
+  Stats::StatName stat_name = stat_name_set_.getStatName(name);
+  return histogram(stat_name);
+}
+
+Stats::Histogram& RedisCommandStats::histogram(Stats::StatName stat_name) {
+  const Stats::SymbolTable::StoragePtr stat_name_storage = addPrefix(stat_name);
+  return scope_.histogramFromStatName(Stats::StatName(stat_name_storage.get()));
+}
+
+Stats::CompletableTimespanPtr
+RedisCommandStats::createCommandTimer(std::string name, Envoy::TimeSource& time_source) {
+  Stats::StatName stat_name = stat_name_set_.getStatName(name + latency_suffix_);
+  return std::make_unique<Stats::TimespanWithUnit<std::chrono::microseconds>>(histogram(stat_name),
+                                                                              time_source);
+}
+
+Stats::CompletableTimespanPtr
+RedisCommandStats::createAggregateTimer(Envoy::TimeSource& time_source) {
+  return std::make_unique<Stats::TimespanWithUnit<std::chrono::microseconds>>(
+      histogram(upstream_rq_time_), time_source);
+}
+
+std::string RedisCommandStats::getCommandFromRequest(const RespValue& request) {
+  // Get command from RespValue
+  switch (request.type()) {
+  case RespType::Array:
+    return getCommandFromRequest(request.asArray().front());
+  case RespType::Integer:
```

Comment:
I don't think we should use the integer as a command, this can potentially explode the number of metrics if there's a bug.  Let's limit it to explicitly only strings and array with the first element as a string.  Otherwise, just either skip the metric or put it as something like "unknown".

Reply:
I'll use unknown